src/libponyrt/sched/scheduler.c

```diff
@@ -491,3 +501,69 @@ PONY_API pony_ctx_t* pony_ctx()
   pony_assert(this_scheduler != NULL);
   return &this_scheduler->ctx;
 }
+
+void ponyint_sched_mute(pony_ctx_t* ctx, pony_actor_t* sender, pony_actor_t* recv)
+{
+  scheduler_t* sched = ctx->scheduler;
+  size_t index = HASHMAP_UNKNOWN;
+  muteref_t key;
+  key.key = recv;
+
+  muteref_t* mref = ponyint_mutemap_get(&sched->mute_mapping, &key, &index);
+  if(mref == NULL)
+  {
+    mref = ponyint_muteref_alloc(recv);
+    ponyint_mutemap_putindex(&sched->mute_mapping, mref, index);
+  }
+
+  size_t index2 = HASHMAP_UNKNOWN;
+  pony_actor_t* r = ponyint_muteset_get(&mref->value, sender, &index2);
+  if(r == NULL)
+  {
+    ponyint_muteset_putindex(&mref->value, sender, index2);
+    sender->muted += 1;
```

Comment:
This is currently equivalent to `atomic_fetch_add_explicit(&sender->muted, 1, memory_order_seq_cst)`, i.e. a very expensive operation.

As far as I can see only one scheduler can mute/unmute a given actor at a time, so this can be replaced with

```c
uint8_t muted = atomic_load_explicit(&sender->muted, memory_order_relaxed);
atomic_store_explicit(&sender->muted, muted + 1, memory_order_relaxed);
```

If I'm wrong and multiple schedulers can modify the `muted` field of an actor at the same time, this should be `atomic_fetch_add_explicit(&sender->muted, 1, memory_order_relaxed)` instead.

Reply:
that's correct. only a single scheduler can mute an actor at a time as it happens on message send.